src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -275,19 +250,11 @@ public void OnStopActivity(Activity activity, object payload)
                     activity.SetStatus(SpanHelper.ResolveSpanStatusForHttpStatusCode(activity.Kind, (int)response.StatusCode));
```

Comment:
Could you hardcode the kind to `ActivityKind.Client` here as well similar to how you have done it for the webrequest one?

Reply:
in web request we cannot rely on activity being non-null always. here we can, but I can update this one as well later.